### ISSUE TITLE
`twisted.cred.credentials.DigestCredentialFactory._generateOpaque` should return output with no newlines in it

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3693]
|[<img alt="washort's avatar" src="https://avatars.githubusercontent.com/u/947487?s=50" width="50" height="50">](https://github.com/washort)| @washort reported|
|-|-|
|Trac ID|trac#3693|
|Type|defect|
|Created|2009-03-18 19:52:59Z|
|Branch|https://github.com/twisted/twisted/tree/no-newline-opaque-3693|

It calls `str.encode("base64")` but fails to deal with cases where the output is split into multiple lines.

<details><summary>Searchable metadata</summary>

```
trac-id__3693 3693
type__defect defect
reporter__washort washort
priority__normal normal
milestone__ 
branch__branches_no_newline_opaque_3693 branches/no-newline-opaque-3693
branch_author__washort washort
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1237405979000000 1237405979000000
changetime__1261410773000000 1261410773000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="washort's avatar" src="https://avatars.githubusercontent.com/u/947487?s=50" width="50" height="50">](https://github.com/washort)<a name="note_1"></a>|@washort commented|
|-|-|

(In [26419]) Branching to 'no-newline-opaque-3693'

**Comment by twisted-trac** [comment]
|[<img alt="washort's avatar" src="https://avatars.githubusercontent.com/u/947487?s=50" width="50" height="50">](https://github.com/washort)|@washort removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @washort<br>@exarkun set status to `new`|
|-|-|

  1. twisted/cred/credentials.py
    1. The indentation of the closing quote line for the docstring of `_generateOpaque` got changed somehow.
  1. twisted/test/test_digestauth.py
    1. update copyright date
    1. two lines between methods
    1. The blank line at the beginning of `test_noNewlineOpaque` is annoying. ;)
    1. `assertNotIn` is preferred over `failIf(x in y)`

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess commented|
|-|-|

It looks like exarkun's comments were addressed and 'review' was just never set - merging now.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_6"></a>|@jesstess set status to `closed`|
|-|-|

(In [27842]) Merge no-newline-opaque-3693

Author: washort
Reviewers: exarkun, jesstess
Fixes: [#3693](https://github.com/twisted/twisted/issues/3693)

DigestCredentialFactory._generateOpaque returns a value without
newlines, regardless of the length of the nonce.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
